pkg/tsdb/azuremonitor/azuremonitor-resource-handler.go

```diff
@@ -33,30 +33,31 @@ func (s *httpServiceProxy) writeErrorResponse(rw http.ResponseWriter, statusCode
 	rw.Header().Set("Content-Type", "application/json")
 	rw.WriteHeader(statusCode)
 
+	// Set error response to initial error message
+	errorBody := map[string]string{"error": message}
+
 	// Attempt to locate JSON portion in error message
 	re := regexp.MustCompile(`\{.*?\}`)
 	jsonPart := re.FindString(message)
-
-	var jsonData map[string]interface{}
-	if unmarshalErr := json.Unmarshal([]byte(jsonPart), &jsonData); unmarshalErr != nil {
-		errorMsg, _ := json.Marshal(map[string]string{"error": "Invalid JSON format in error message"})
-		_, err := rw.Write(errorMsg)
-		if err != nil {
-			return fmt.Errorf("unable to write HTTP response: %v", err)
+	if jsonPart != "" {
+		var jsonData map[string]interface{}
+		if unmarshalErr := json.Unmarshal([]byte(jsonPart), &jsonData); unmarshalErr != nil {
+			errorBody["error"] = fmt.Sprintf("Invalid JSON format in error message. Raw error: %s", message)
```

Comment:
shouldn't this always happen even if we don't find json?

Reply:
No, if we don't find JSON there may still be a valid string error message returned.